src/marshmallow/utils.py

```diff
@@ -197,7 +197,10 @@ def from_timestamp(value: typing.Any) -> dt.datetime:
 
     # Load a timestamp with utc as timezone to prevent using system timezone.
     # Then set timezone to None, to let the Field handle adding timezone info.
-    return dt.datetime.fromtimestamp(value, tz=dt.timezone.utc).replace(tzinfo=None)
+    try:
+        return dt.datetime.fromtimestamp(value, tz=dt.timezone.utc).replace(tzinfo=None)
+    except (OSError, OverflowError) as exc:
+        raise ValueError("Timestamp is too large") from exc
```

Comment:
The message is fine for overflow but probably wrong for OSError, right?

> [fromtimestamp()](https://docs.python.org/3/library/datetime.html#datetime.datetime.fromtimestamp) may raise [OverflowError](https://docs.python.org/3/library/exceptions.html#OverflowError), if the timestamp is out of the range of values supported by the platform C localtime() or gmtime() functions, and [OSError](https://docs.python.org/3/library/exceptions.html#OSError) on localtime() or gmtime() failure.

We should use a more generic message to include `OSError`. Or use two different messages, but the message doesn't matter that much since it is not used to build the `ValidationError`.

Reply:
I changed it to two differents messages.